### PR TITLE
cities index page

### PR DESCRIPTION
**Pull request by base-p**
this is just an implementation for the cities index page. i had to tamper with the environments file we can roll it back later. couldnt find where to change it on the fly. if this is approved I can move on to cities show page

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
         "cordova-plugin-whitelist": "^1.3.3",
         "core-js": "^2.5.4",
         "g": "^2.0.1",
+        "node-sass": "^4.13.1",
```
Comment (piousbox) [inline package.json]: You aren't using sass here, do you need this dependency?

File: src/app/CitiesComponent/CitiesIndex.page.ts [hunk 2]
```diff
@@ -1,25 +1,66 @@
...
-  ) {}
+    public toastController: ToastController,
+  ) {
+      const answer = this.httpClient.get(ApiRouter.citiesindex)
+    answer.subscribe(data => {
+      if (data) {
+        this.cities = data;
+      }
+    }, async error => {
+      console.log('+++ citiesindex 1:', JSON.stringify(error))
```
Comment (piousbox) [inline src/app/CitiesComponent/CitiesIndex.page.ts]: you don't need to stringify in console, by the way

File: src/environments/environment.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
 export const environment = {
   name: 'environment',
-  production: false,
+  production: true,
```
Comment (piousbox) [inline src/environments/environment.ts]: production = false, there is a specific environment that is a production environment. Why did you change this?

File: src/app/CitiesComponent/CitiesIndex.page.ts [hunk 4]
```diff
@@ -1,25 +1,66 @@
...
+        duration: 2000
+      });
+      toast.present();
+    });
+  }
+    
+render(){
+    const answer = this.httpClient.get(ApiRouter.citiesindex)
+    answer.subscribe(data => {
+      if (data['newsitems']) {
```
Comment (piousbox) [inline src/app/CitiesComponent/CitiesIndex.page.ts]: so... you're expecting to get newsitems from https://manager.piousbox.com/api/cities.json ?

File: src/app/CitiesComponent/CitiesIndex.page.ts [hunk 5]
```diff
@@ -1,25 +1,66 @@
...
+}    
   navigate(where) {
     this.router.navigate([where]);
   }
-  ngOnInit() {}
+    
+  ngOnInit() {
+      this.appService.currentMessage.subscribe( message => {
+      console.log('+++ new message:', message)
+      if (message == C.didLogin) {
```
Comment (piousbox) [inline src/app/CitiesComponent/CitiesIndex.page.ts]: but cities display even to non-logged-in users, so what does this do?

File: src/app/CitiesComponent/CitiesIndex.page.ts (hunk 2 above)
Comment (base-p) [reply src/app/CitiesComponent/CitiesIndex.page.ts]: i know

File: src/app/CitiesComponent/CitiesIndex.page.ts (hunk 5 above)
Comment (base-p) [reply src/app/CitiesComponent/CitiesIndex.page.ts]: I will fix this. i thought cities would be for only logged in users. I implemented this twice anyways

File: src/app/CitiesComponent/CitiesIndex.page.ts (hunk 4 above)
Comment (base-p) [reply src/app/CitiesComponent/CitiesIndex.page.ts]: i will fix this

File: package.json (hunk 1 above)
Comment (base-p) [reply package.json]: ' ionic cordova run browser --livereload --address tgm.mac --ssl --port 8100' wouldnt run on my machine without node-sass. i get this error "[ng] Error: Cannot find module 'node-sass'"

File: src/environments/environment.ts (hunk 3 above)
Comment (base-p) [reply src/environments/environment.ts]: ok.
